lexnlp/ml/model_io.py

```
@@ -0,0 +1,123 @@
+
+def _load_legacy(path: Path) -> Any:
+    """Load a legacy pickle / cloudpickle / joblib artifact.
+
+    Legacy loaders are imported lazily because ``cloudpickle`` and
+    ``joblib`` are optional for skops-only consumers.
+    """
+
+    suffix = path.suffix.lower()
+    if suffix in (".pickle", ".pkl"):
+        with path.open("rb") as file:
+            return pickle.load(file)
```

Ruff (0.15.10)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)